wallet/src/repository/wallet/mod.rs

```diff
@@ -546,14 +562,20 @@ where
             let model::Path {
                 keychain, index, ..
             } = self.db.get(&keys::pkh(&key_balance.pkh))?;
-            let parent_key = &state.keychains[keychain as usize];
+            let parent_key = &state
+                .keychains
+                .get(keychain as usize)
+                .expect("could not get keychain");
 
             let extended_sign_key =
                 parent_key.derive(&self.engine, &types::KeyPath::default().index(index))?;
 
             payment = payment
                 .checked_add(key_balance.amount)
                 .ok_or_else(|| Error::TransactionValueOverflow)?;
+            balance = balance
+                .checked_sub(key_balance.amount)
+                .ok_or_else(|| Error::BalanceUnderflow)?;
```

Comment:
Should the balance be updated when the transaction is created, or when the transaction is included in a block?

Note that in the future we will have different types of balance: confirmed and pending.

Reply:
Since the used UTXOs cannot be used to create other transactions, in practice the balance has dropped down, so I would leave it as it is at the moment